code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -187,113 +230,165 @@
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "lay_web"
+	icon_icon = 'icons/mob/actions/actions_animal.dmi'
+	background_icon_state = "bg_alien"
 
-	var/T = src.loc
-
-	if(stat == DEAD)
+/datum/action/innate/lay_web/Activate()
+	if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse))
 		return
-	if(busy != SPINNING_WEB)
-		busy = SPINNING_WEB
-		src.visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance.</span>")
-		stop_automated_movement = 1
-		if(do_after(src, 40, target = T))
-			if(busy == SPINNING_WEB && src.loc == T)
-				new /obj/structure/spider/stickyweb(T)
-		busy = SPIDER_IDLE
-		stop_automated_movement = FALSE
+	var/mob/living/simple_animal/hostile/poison/giant_spider/nurse/S = owner
 
+	if(!isturf(S.loc))
+		return
+	var/turf/T = get_turf(S)
 
-/mob/living/simple_animal/hostile/poison/giant_spider/nurse/verb/Wrap()
-	set name = "Wrap"
-	set category = "Spider"
-	set desc = "Wrap up prey to feast upon and objects for safe keeping."
+	var/obj/structure/spider/stickyweb/W = locate() in T
+	if(W)
+		to_chat(S, "<span class='warning'>There's already a web here!</span>")
+		return
 
-	if(stat == DEAD)
+	if(S.busy != SPINNING_WEB)
+		S.busy = SPINNING_WEB
+		S.visible_message("<span class='notice'>[S] begins to secrete a sticky substance.</span>","<span class='notice'>You begin to lay a web.</span>")
+		S.stop_automated_movement = TRUE
+		if(do_after(S, 40, target = T))
+			if(S.busy == SPINNING_WEB && S.loc == T)
+				new /obj/structure/spider/stickyweb(T)
+		S.busy = SPIDER_IDLE
+		S.stop_automated_movement = FALSE
+	else
+		to_chat(S, "<span class='warning'>You're already spinning a web!</span>")
+
+/datum/action/innate/wrap
+	name = "Wrap"
+	desc = "Wrap something or someone in a cocoon. If it's a living being, you'll also consume them, allowing you to lay eggs."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "wrap"
+	icon_icon = 'icons/mob/actions/actions_animal.dmi'
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/wrap/Activate()
+	if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse))
 		return
-	if(!cocoon_target)
+	var/mob/living/simple_animal/hostile/poison/giant_spider/nurse/S = owner
+
+	if(!S.cocoon_target)
 		var/list/choices = list()
-		for(var/mob/living/L in view(1,src))
-			if(L == src || L.anchored)
+		for(var/mob/living/L in view(1,S))
+			if(L == S || L.anchored)
 				continue
 			if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
 				continue
-			if(Adjacent(L))
+			if(S.Adjacent(L))
 				choices += L
-		for(var/obj/O in src.loc)
+		for(var/obj/O in S.loc)
 			if(O.anchored)
 				continue
-			if(Adjacent(O))
+			if(S.Adjacent(O))
 				choices += O
-		var/temp_input = input(src,"What do you wish to cocoon?") in null|choices
-		if(temp_input && !cocoon_target)
-			cocoon_target = temp_input
+		var/temp_input = input(S,"What do you wish to cocoon?") in null|choices

```

Comment:
now you need to check your choice still exists

also this stuff is bad it should be click actions not input choices, however that's not really your fault

Reply:
oh I missed you did do it below, ignore this comment